internal/engine/compiler/impl_arm64.go

```diff
@@ -2642,17 +2641,16 @@ func (c *arm64Compiler) compileLoad8(o *wazeroir.OperationLoad8) error {
 	var vt runtimeValueType
 	switch o.Type {
 	case wazeroir.SignedInt32:
-		// TODO used 32-bit variant.
-		loadInst = arm64.MOVB
+		loadInst = arm64.LDRSBW
```

Comment:
this and below for i32.load16_s are the fix for arm64 compiler. 

Other changes are mechanical according to the asm package refactoring.